[Writing Tools] Adopt new names for NS/UIWritingToolsAllowedInputOptions

rr-codes opened the pull request:
#### 785069e5ddf2eedf46d6c7aeabe4ff61b60fc918
<pre>
[Writing Tools] Adopt new names for NS/UIWritingToolsAllowedInputOptions
<a href="https://bugs.webkit.org/show_bug.cgi?id=275935">https://bugs.webkit.org/show_bug.cgi?id=275935</a>
<a href="https://rdar.apple.com/130540106">rdar://130540106</a>

Reviewed by Tim Horton and Abrar Rahman Protyasha.

Temporarily add staging declarations for the new API and remove uses of the old API, while still
retaining equal functionality for both APIs.

* Source/WebCore/PAL/pal/spi/cocoa/WritingToolsSPI.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView allowedWritingToolsResultOptions]):
(-[WKWebView writingToolsAllowedInputOptions]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView writingToolsAllowedInputOptions]):
(-[WKContentView allowedWritingToolsResultOptions]):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingTools.mm:
(-[WritingToolsWKWebView allowedWritingToolsResultOptionsForTesting]):
(TEST(WritingTools, AllowedInputOptionsNonEditable)):
(TEST(WritingTools, AllowedInputOptionsEditable)):
(TEST(WritingTools, AllowedInputOptionsRichText)):
(TEST(WritingTools, AllowedInputOptionsPlainText)):
(-[WritingToolsWKWebView writingToolsAllowedInputOptionsForTesting]): Deleted.

Canonical link: <a href="https://commits.webkit.org/280945@main">https://commits.webkit.org/280945@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/29e063be0b5fc72547094b5bccefc6681b4982b0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/58101 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/37429 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/10581 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/61726 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/8546 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/60229 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/45065 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/8737 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/47072 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/6087 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/60131 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/35094 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/50226 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/27906 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/31857 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/7523 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/7550 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/53804 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/7791 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/63427 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/2015 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/7850 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/54341 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/2022 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/50239 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/54446 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/1724 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8670 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/33258 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/34344 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/35428 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/34089 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
```diff
@@ -13186,9 +13186,15 @@ - (void)systemDidChangeSelectionForInteraction:(BETextInteraction *)interaction
+    return [_webView allowedWritingToolsResultOptions];
+}
+
+- (UIWritingToolsAllowedInputOptions)allowedWritingToolsResultOptions
```
I think this might end up being a build failure once UIKit updates.

rr-codes replied:
fixed, and verified it builds fine across all configurations

hortont424 reviewed (approved):
IMO you could sprinkle the fixmes a little less liberally

aprotyas commented on Source/WebCore/PAL/pal/spi/cocoa/WritingToolsSPI.h:
```diff
@@ -41,12 +41,14 @@ constexpr auto PlatformWritingToolsBehaviorDefault = NSWritingToolsBehaviorDefau
-constexpr auto PlatformWritingToolsAllowedInputOptionsRichText = NSWritingToolsAllowedInputOptionsRichText;
-constexpr auto PlatformWritingToolsAllowedInputOptionsList = NSWritingToolsAllowedInputOptionsList;
-constexpr auto PlatformWritingToolsAllowedInputOptionsTable = NSWritingToolsAllowedInputOptionsTable;
+using PlatformWritingToolsResultOptions = NSUInteger;
+
+constexpr auto PlatformWritingToolsResultPlainText = (PlatformWritingToolsResultOptions)(1 << 0);
+constexpr auto PlatformWritingToolsResultRichText = (PlatformWritingToolsResultOptions)(1 << 1);
+constexpr auto PlatformWritingToolsResultList = (PlatformWritingToolsResultOptions)(1 << 2);
+constexpr auto PlatformWritingToolsResultTable = (PlatformWritingToolsResultOptions)(1 << 3);
```
It looks like we can move this outside of the platform-specific paths? Since it's the same declaration in both cases.

rr-codes replied:
I'd rather not, since once I can remove the old symbols, all of this will just go back to being platform-dependent anyways, so less code will required to be changed.

aprotyas replied:
Oh okay, I didn't realize that. Np!

aprotyas commented on Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
```diff
@@ -407,9 +407,13 @@ struct PerWebProcessState {
 #if PLATFORM(MAC)
-- (NSWritingToolsAllowedInputOptions)writingToolsAllowedInputOptions;
+// FIXME: (rdar://130540028) Remove uses of the old WritingToolsAllowedInputOptions API in favor of the new WritingToolsResultOptions API, and remove staging.
+- (PlatformWritingToolsResultOptions)writingToolsAllowedInputOptions;
+- (PlatformWritingToolsResultOptions)allowedWritingToolsResultOptions;
 #else
-- (UIWritingToolsAllowedInputOptions)writingToolsAllowedInputOptions;
+// FIXME: (rdar://130540028) Remove uses of the old WritingToolsAllowedInputOptions API in favor of the new WritingToolsResultOptions API, and remove staging.
+- (PlatformWritingToolsResultOptions)writingToolsAllowedInputOptions;
+- (PlatformWritingToolsResultOptions)allowedWritingToolsResultOptions;
```
Same thing here, since they're identical, we can avoid the platform-specific declarations, right?

rr-codes committed:
[Writing Tools] Adopt new names for NS/UIWritingToolsAllowedInputOptions
https://bugs.webkit.org/show_bug.cgi?id=275935
rdar://130540106

Reviewed by Tim Horton and Abrar Rahman Protyasha.

Temporarily add staging declarations for the new API and remove uses of the old API, while still
retaining equal functionality for both APIs.

* Source/WebCore/PAL/pal/spi/cocoa/WritingToolsSPI.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView allowedWritingToolsResultOptions]):
(-[WKWebView writingToolsAllowedInputOptions]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView writingToolsAllowedInputOptions]):
(-[WKContentView allowedWritingToolsResultOptions]):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingTools.mm:
(-[WritingToolsWKWebView allowedWritingToolsResultOptionsForTesting]):
(TEST(WritingTools, AllowedInputOptionsNonEditable)):
(TEST(WritingTools, AllowedInputOptionsEditable)):
(TEST(WritingTools, AllowedInputOptionsRichText)):
(TEST(WritingTools, AllowedInputOptionsPlainText)):
(-[WritingToolsWKWebView writingToolsAllowedInputOptionsForTesting]): Deleted.

Canonical link: https://commits.webkit.org/280945@main
